cmake/01_version.cmake

```diff
@@ -42,8 +42,8 @@ if(error_version_check)
 endif()
 
 # Extract major, minor and patch versions. NOTE: works only for repo tag in format vX.Y.Z
-if(NOT ${REPO_TAG} MATCHES "^v([0-9]+)\\.([0-9]+)\\.([0-9]+)$")
-  message(FATAL_ERROR "Expected version tag in format v1.2.3, got ${REPO_TAG}")
+if(NOT ${REPO_TAG} MATCHES "^v([0-9]+)\\.([0-9]+)\\.([0-9]+)(-.+)?$")
+  message(FATAL_ERROR "Expected version tag in format v1.2.3 or v1.2.3-*, got ${REPO_TAG}")
```

Comment:
this was needed, because this version check is tripped up by our v1.2.3-rc1 etc. tags